lib/functoria/device.mli

```diff
@@ -51,6 +51,9 @@ val pp : 'b Fmt.t -> ('a, 'b) t Fmt.t
 val equal : ('a, 'b) t -> ('c, 'd) t -> bool
 (** [equal] is the equality function for devices. *)
 
+val witness : ('a, _) t -> ('b, _) t -> ('a, 'b) Misc.Eq.eq
```

Comment:
Maybe move `Eq` to a top-level module, that's not very nice to have `Misc` leaking in the public API.

Reply:
Done, it's called `Typeid`